src/twisted/internet/test/test_core.py

```diff
@@ -30,10 +34,11 @@ def assertFullyNewStyle(self, instance):
         This is a beneficial condition because PyPy is better able to
         optimize attribute lookup on such classes.
         """
-        self.assertIsInstance(instance, object)
+        testCase = cast(SynchronousTestCase, self)
+        testCase.assertIsInstance(instance, object)
```

Comment:
```suggestion
        testCase = cast(SynchronousTestCase, self)
        testCase.assertIsInstance(instance, object)
```

rather than a cast you can use a mixin self type: https://mypy-play.net/?mypy=latest&python=3.10&flags=show-error-codes%2Cstrict&gist=686e97040538b72c2b9d0a1f73aed209

Reply:
but I think it would be better to just delete this method or replace it with:

```python
def assertFullyNewStyle(self, *args: object, **kwargs: object) -> None:
    pass
```

because it cannot fail

Reply:
https://twistedmatrix.com/trac/ticket/10277#ticket